src/utils/recommendationUtility.js

```diff
@@ -0,0 +1,40 @@
+const SpotifyWebApi = require('spotify-web-api-node'); // Import Spotify Web API library
+const AlbumRecommendation = require('../models/AlbumRecommendation');
+
+// Set up Spotify Web API client with credentials
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.SPOTIFY_CLIENT_ID,
+  clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+});
+
+async function generateRecommendations(userId, listeningHistory) {
+  try {
+    // Check if listeningHistory is empty
+    if (listeningHistory.length === 0) {
+      console.warn('No listening history found for user:', userId);
+      return;
+    }
+
+    // Extract artistIds from listeningHistory
+    const artistIds = listeningHistory.map((entry) => entry.artistId);
```

Comment:
Currently, we aren't storing the spotify artist id in our application. So we will need to first make a call to Spotify to get the artist id. 

There is a `Get Album` by id endpoint: [https://developer.spotify.com/documentation/web-api/reference/get-an-album](https://developer.spotify.com/documentation/web-api/reference/get-an-album)

Our spotify-web-api library that we're using can access it with the `getAlbum` function: [https://github.com/thelinmichael/spotify-web-api-node#more-examples](https://github.com/thelinmichael/spotify-web-api-node#more-examples)

The album id we need is the _spotify album id_ and not our own Mongo object id. Right now, the only place we're storing that is in the `spotifyUrl`, we may want to change that later but for now, we can follow a similar pattern as the frontend where we use regex to extract that id from the url field: [https://github.com/Code-the-Dream-School/dd-prac-team4-front/blob/40ffb979b0229c5f2485eaecc87e31af55fd3fa4/src/components/album/AlbumPreview.js#L19-L29](https://github.com/Code-the-Dream-School/dd-prac-team4-front/blob/40ffb979b0229c5f2485eaecc87e31af55fd3fa4/src/components/album/AlbumPreview.js#L19-L29)

An album can have multiple artists, for now just grab the first artist id in the returned list.